File: app/assets/javascripts/controllers/profiles_controller.js
App.ProfilesController = Ember.ArrayController.extend({
  itemController: 'profile',
  sortProperties: ['id'],
  actions: {
    createProfile: function() {
      var controller = this;
      var name = this.get('newName') || '';
      var email = this.get('newEmail') || '';
      var upworkName = this.get('newUpwork') || '';
      var message = this.get('newMessage') || '';
      $.post('/api/profiles.json', { profile: { name: name, email: email, message: message, upwork_name: upworkName } })
      .done( function(response) {
        controller.store.push('profile', response.profile);
        controller.set('newName', '');
        controller.set('newEmail', '');
        controller.set('newUpwork', '');
        controller.set('newMessage', '');
      })
      .fail( function(xhr, textStatus, errorThrown) {
        var errors = $.parseJSON(xhr.responseText).errors
        controller.set('errors', errors);
        controller.send('displayErrors');
      });
    }
  }
});

File: app/controllers/api/profiles_controller.rb
class Api::ProfilesController < Api::BaseController
  def index
    profiles = Profile.all
    respond_with profiles
  end

  def show
    profile = Profile.find(params[:id])
    respond_with profile
  end

  def create
    profile = Profile.create(profile_params)
    respond_with :api, profile
  end

  def update
    profile = Profile.find(params[:id])
    profile.update(profile_params)
    respond_with profile
  end

  def destroy
    profile = Profile.find(params[:id])
    respond_with profile.destroy
  end

  def profile_params
    params.require(:profile).permit(:name, :email, :message, :upwork_name)
  end
end
